cdk/domino_cdk/provisioners/eks/eks_iam_roles_for_k8s.py

```diff
@@ -0,0 +1,176 @@
+        "ecr:DescribeImageScanFindings",
+        "ecr:GetLifecyclePolicyPreview",
+        "ecr:CreateRepository",
+        "ecr:PutImageScanningConfiguration",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:DeleteLifecyclePolicy",
+        "ecr:PutImage",
+        "ecr:UntagResource",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:StartLifecyclePolicyPreview",
+        "ecr:InitiateLayerUpload",
+        "ecr:GetRepositoryPolicy",
+    ]
+}
+
+s3_policies = {
+    "write_blobs_read_logs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+        "logs": s3_read_permissions,
+    },
+    "write_blobs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+    },
+    "read_logs": {
+        "logs": s3_read_permissions,
+    },
+    "read_blobs": {
+        "blobs": s3_read_permissions,
+    },
+}
+
+
+# Roles
+roles = {
+    "nucleus": [
+        "write_blobs_read_logs",
+    ],
+    "executor": [
+        "write_blobs",
+    ],
+    "builder": [
+        "ecr_read_write_permissions",
+    ],
+}
+
+
+class DominoEksK8sIamRolesProvisioner:
+    def __init__(
+        self,
+        scope: cdk.Construct,
+    ) -> None:
+        self.scope = scope
+
+    def print_children(self, construct):
+        print(construct.node.path, construct.to_string())
+        for c in construct.node.children:
+            self.print_children(c)
+
+    def provision(self, stack_name: str, cluster: eks.Cluster, buckets: Dict[str, Bucket]):
+        # Create dummy service account just to make EKS to do heavy listing of creating OIDC
+        # TODO: perform associate_iam_oidc_provider by lambda so we do  not create dummy stuff
+        sa = eks.ServiceAccount(cluster, "dummy", cluster=cluster, name="dummy")
+        # Then we copy policy doc
+        statement_json = sa.role.assume_role_policy.to_json()['Statement'][0]
+        del statement_json['Condition']['StringEquals']
+        logical_id = self.scope.get_logical_id(
+            # Warning! Magic! We go two levels deep because child OpenIdConnectProvider is type Construct but
+            # logical_id belongs to type CfnElement, which is linked from below like this.
+            cluster.node.find_child("OpenIdConnectProvider").node.default_child.node.default_child
+        )
+        fn = cdk.Fn.select(
+            1,
+            cdk.Fn.split(
+                ":oidc-provider/",
+                cdk.Fn.ref(logical_id),
+            ),
+        )
+        statement_json['Condition']['StringLike'] = cdk.CfnJson(
+            self.scope, "OidcJson", value={f"{fn}:aud": "sts.amazonaws.com", f"{fn}:sub": "system:serviceaccount:*"}
+        )
+        managed_policies = {}
+        for name, cfg in ecr_policies.items():
+            managed_policies[name] = self.create_ecr_policy(stack_name, name, cfg)
+        for name, cfg in s3_policies.items():
+            managed_policies[name] = self.create_s3_policy(name, name, cfg, buckets)
```

Comment:
Is the first `name` supposed to be `stack_name`?

Reply:
No, it is policy name.

But create_s3_policy had unused second parameter. Thanks.